include/aie/Dialect/AIEVec/IR/AIEVecOps.td

```diff
@@ -312,20 +312,40 @@ def AIEVec_MulOp:
 }
 
 def AIEVec_MulElemOp:
-  AIEVec_Op<"mul_elem", [
-    Pure
-  ]>,
-  Arguments<(ins AnyVector:$lhs, AnyVector:$rhs)>,
-  Results<(outs AnyVector:$result)> {
+  AIEVec_Op<"mul_elem", [	 
+    Pure,
+    SameTypeOperands,
+    SameOperandsShape,
+    IsValidAccTypeForAIE2MulElem<"lhs", "rhs", "result">
+  ]>,	  
+  Arguments<(ins
+    VectorOfLengthAndType<[16, 32, 64], [I8, I16, I32, BF16, F32]>:$lhs, 
+    VectorOfLengthAndType<[16, 32, 64], [I8, I16, I32, BF16, F32]>:$rhs)>,
+  Results<(outs
+    VectorOfLengthAndType<[16, 32], [I8, I16, I32, I64, BF16, F32]>:$result)> {
   let summary = "AIE-ML vector element-wise multiply";
   let description = [{
     AMD-specific multiply operation that multiplies two 1-D vectors in the same channel.
     The vector sizes are at least 512 bits.
     `$result = `$lhs * $rhs`.
+    Currently, the following are the supported type combinations:
+        lhs                | rhs                | Accumulator
+      :------------------:|:------------------:|:-----------------:
+        `vector<32xi8>`    | `vector<32xi8>`    | `vector<32xi8>`
+        `vector<32xi8>`    | `vector<32xi8>`    | `vector<32xi32>`
+        `vector<64xi8>`    | `vector<64xi8>`    | `vector<32xi32>`
+        `vector<32xi16>`   | `vector<32xi16>`   | `vector<32xi16>`
+        `vector<32xi16>`   | `vector<32xi16>`   | `vector<32xi32>`
+        `vector<32xi16>`   | `vector<32xi16>`   | `vector<16xi64>`
+        `vector<16xi32>`   | `vector<16xi32>`   | `vector<16xi32>`
+        `vector<16xi32>`   | `vector<16xi32>`   | `vector<16xi64>`
+        `vector<16xbf16>`  | `vector<16xbf16>`  | `vector<16xbf16>`
+        `vector<16xbf16>`  | `vector<16xbf16>`  | `vector<16xf32>`
+        `vector<16xf32>`   | `vector<16xf32>`   | `vector<16xf32>`'
```

Comment:
We should just have:
```
        `vector<32xi8>`    | `vector<32xi8>`    | `vector<32xi32>`
        `vector<32xi16>`   | `vector<32xi16>`   | `vector<32xi32>`
        `vector<16xi32>`   | `vector<16xi32>`   | `vector<16xi64>`
        `vector<16xbf16>`  | `vector<16xbf16>`  | `vector<16xf32>`
        `vector<16xf32>`   | `vector<16xf32>`   | `vector<16xf32>`
```
These operations refer to the following intrinsics in the table at https://www.xilinx.com/htmldocs/xilinx2023_2/aiengine_ml_intrinsics/intrinsics/group__intr__gpvectorop__mul.html. 
```
8-bit x 8-bit = 32-bit,	32,	1x2,	2x1,	1x1
16-bit x 16-bit = 32-bit,	32,	1x1,	1x1,	1x1
bfloat16 x bfloat16 = fp32,	16,	1x2,	2x1,	1x1
```
The two more are at https://www.xilinx.com/htmldocs/xilinx2023_2/aiengine_ml_intrinsics/intrinsics/group__intr__gpvectorop__emul__32bx32b.html#gae6f04990833f0cf4b62bf1433ac11e6a and 
https://www.xilinx.com/htmldocs/xilinx2023_2/aiengine_ml_intrinsics/intrinsics/group__intr__gpvectorop__mul__fp32xfp32.html#gac1e9a132ee05edf8b6afe9c985598768

We can support `arith.muli/arith.mulf` in different result types (as you listed above), because we add `aievec.cast/aievec.srs` to cast the acc types back to normal integer/floating types. However, that shouldn't be part of the semantic of `aievec.mul_elem`.